client/src/featureform/register.py

```diff
@@ -2599,6 +2599,16 @@ def register_gcs(
                 has all the functionality of OfflineProvider
         """
         tags, properties = set_tags_properties(tags, properties)
+
+        if bucket_name == "":
+            raise ValueError("bucket_name is required and cannot be empty string")
+
+        bucket_name = bucket_name.replace("gs://", "")
```

Comment:
all this logic feels like it should belong somewhere else.

like FileStore.S3.validate_bucket_name()

similarly with s3 and azure

Reply:
it's minor but encapsulating the logic means it can be reusable and also have a dedicated place for it